pkg/middlewares/metrics/metrics.go

```diff
@@ -78,23 +74,25 @@ func WrapServiceHandler(ctx context.Context, registry metrics.Registry, serviceN
 }
 
 func (m *metricsMiddleware) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	labels := []string{"method", getMethod(req), "protocol", getRequestProtocol(req)}
+	// Initialize labels slice with correct size -> 6 + len(m.baseLabels)
+	labels := make([]string, 0, 6+len(m.baseLabels))
```

Comment:
```suggestion
	labels := make([]string, 0, 2*len([]string{"method", "protocol", "code"})+len(m.baseLabels))
```
then it is self-documented, and we'll never make the mistake of adding new labels while forgetting to update that hardcoded 6.

Reply:
👍